shell/renderer/atom_render_frame_observer.cc

```diff
@@ -68,16 +68,25 @@ void AtomRenderFrameObserver::DidCreateScriptContext(
   if (ShouldNotifyClient(world_id))
     renderer_client_->DidCreateScriptContext(context, render_frame_);
 
+  auto* command_line = base::CommandLine::ForCurrentProcess();
+
   bool use_context_isolation = renderer_client_->isolated_world();
+  // This logic matches the EXPLAINED logic in atom_renderer_client.cc
+  // to avoid explaining it twice go check that implementation in
+  // DidCreateScriptContext();
   bool is_main_world = IsMainWorld(world_id);
   bool is_main_frame = render_frame_->IsMainFrame();
```

Comment:
The overall logic matches though I think 🤔

Reply:
that's far from obvious to me :\

Reply:
Yeah the logic isn't great, let alone duplicated, I want to avoid touching it too much in this PR though.  Happy to try come in for a sweeping clean up / dedupe at some point